### PR TITLE
Meta+LibWasm: Infra for wasm js repl

### PR DESCRIPTION
**Pull request by alimpfard**
Makes libjs.dev/repl happen.

TODO:
- [x] Fix recursive parse in LibWasm
    - [x] Check for regressions
- [x] Remove the LibWasm parser stream hack
- [x] Unwip Andrew's commit and remove unrelated bits
- [x] Add a CI job to build this stuff (and possibly deploy to libjs.dev)

### CODE REVIEW

File: Userland/Libraries/LibCore/EventLoop.h [hunk 1]
```diff
@@ -61,6 +61,7 @@ class EventLoop {
     static EventLoop& current();
 
     bool was_exit_requested() const { return m_exit_requested; }
+    int exit_code() const { return m_exit_code; }
```
Comment (ADKaster) [inline Userland/Libraries/LibCore/EventLoop.h:64]: Can ditch this change

File: Userland/Libraries/LibCore/MappedFile.cpp [hunk 2]
```diff
@@ -44,7 +44,11 @@ MappedFile::MappedFile(void* ptr, size_t size)
 
 MappedFile::~MappedFile()
 {
-    MUST(Core::System::munmap(m_data, m_size));
+    auto res = Core::System::munmap(m_data, m_size);
+    if (res.is_error()) {
+        dbgln("cannot unmap file because: {}", res.error());
```
Comment (ADKaster) [inline Userland/Libraries/LibCore/MappedFile.cpp:49]: Fun formatting on this. The fact that munmap is ENOSYS on emscripten is a bit concerning tbh.

File: Userland/Libraries/LibJS/Print.h [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+#include <LibJS/Runtime/Value.h>
+
+namespace JS {
+struct PrintContext {
+    JS::VM& vm;
+    Core::Stream::Stream& stream;
+    bool strip_ansi { false };
+};
+
+ErrorOr<void> print(JS::Value value, PrintContext);
```
Comment (ADKaster) [inline Userland/Libraries/LibJS/Print.h:22]: This signature looks kinda funny. Isn't that print context non copyable and non moveable? How can we pass by value?

File: Meta/Lagom/CMakeLists.txt [hunk 4]
```diff
@@ -474,6 +474,15 @@ if (BUILD_LAGOM)
 
         add_executable(js ../../Userland/Utilities/js.cpp)
         target_link_libraries(js LibCrypto LibJS LibLine LibLocale LibMain LibTextCodec Threads::Threads)
+        if (EMSCRIPTEN)
+            add_executable(libjs ../../Meta/Lagom/Wasm/js_repl.cpp)
```
Comment (ADKaster) [inline Meta/Lagom/CMakeLists.txt:478]: This could just be Wasm/js_repl.cpp. we're in meta/lagom

File: Userland/Libraries/LibJS/Print.h (hunk 3 above)
Comment (alimpfard) [reply Userland/Libraries/LibJS/Print.h:22]: Why would it be non-copyable?

Comment (ADKaster) [reply Userland/Libraries/LibJS/Print.h:22]: Reference data members? Doesn't having them delete all the special members?

Comment (alimpfard) [reply Userland/Libraries/LibJS/Print.h:22]: I'll make it take the context by ref, but I'm still not sure why you expect it to be non-copyable, it's just a bunch of references and a bool - the copy is just memcpy()

Comment (alimpfard) [reply Userland/Libraries/LibJS/Print.h:22]: > Doesn't having them delete all the special members?

only rvalue refs for copy/move ctors, "any" reference type only deletes the default ctor afaik

Comment (alimpfard) [reply Userland/Libraries/LibJS/Print.h:22]: see "Deleted implicitly-declared copy constructor" in  https://en.cppreference.com/w/cpp/language/copy_constructor.

**Review by ADKaster — APPROVED** [review]
lgtm. Bit worried about not pinning emscripten version to something, but if it becomes a problem we'll notice in the CI.

**Pull request merged**